common/src/PactNumber.hs

```diff
@@ -0,0 +1,104 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PackageImports #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE RecursiveDo #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+
+module PactNumber where
```

Comment:
@mightybyte I had set out to apply this suggestion, but while doing that, I've noticed that this `PactNumber` module shared a lot of code with [`chainweb-api`'s `ParsedNumbers`](https://github.com/kadena-io/chainweb-api/blob/953019e5eedc1d23833cb7b5b6c5f53627ff2b20/lib/Chainweb/Api/ParsedNumbers.hs), so I've pushed [a commit](https://github.com/kadena-io/block-explorer/pull/39/commits/2fd8233aa999598efa5ebf1d95689d0c389d3e60) to deduplicate that common code. 

I've noticed that the `FromJSON` instance for `PactNumber` is almost `parseJSON @ParsedInteger <|> parseJSON @ParsedDecimal` but there are minor differences (handling of `String`s, rounding of `Number` etc.), so we probably need to retain this `instance FromJSON PactNumber`.

I should add that I'm not fully satisfied with the current state of affairs. `PactNumber` is sharing things like `integerCodec` and `decimalCodec` with `ParsedNumbers`, but I can't tell whether this sharing is essential or accidental, because we're not defining what a `PactNumber` is and what kind of integers and decimals `integerCodec` and `decimalCodec` are representing. Without pinning down the meaning of these things, it's impossible to tell whether this sharing is essential and if this is an accidental sharing, it's prone to breaking when the meaning of `integerCodec` or `decimalCodec` diverges from `PactNumber`.

So, I suggest one of:
a) Revert [my deduplication commit](https://github.com/kadena-io/block-explorer/pull/39/commits/2fd8233aa999598efa5ebf1d95689d0c389d3e60) so that we don't risk accidental sharing.
b) Pin down the semantics of these things and make sure that the sharing is essential.

Reply:
Emmanuel addressed this by moving PactNumber to chainweb-api while retaining a copy of the shared code. So we can address the semantic concerns later.